modules/Auth/include/ReservationHandler.hpp

```diff
@@ -1,94 +1,335 @@
-     * @return types::reservation::ReservationResult
-     */
-    types::reservation::ReservationResult reserve(int connector, const ConnectorState& state, bool is_reservable,
-                                                  const types::reservation::Reservation& reservation);
-
-    /**
-     * @brief Function tries to cancel reservation with the given \p reservation_id .
-     *
-     * @param reservation_id
-     * @param execute_callback if true, cancel_reservation_callback will be executed
-     * @return int -1 if reservation could not been cancelled, else the id of the connnector
-     */
-    int cancel_reservation(int reservation_id, bool execute_callback);
-
-    /**
-     * @brief Handler that is called when a reservation was started / used.
-     *
-     * @param connector
-     */
-    void on_reservation_used(int connector);
-
-    /**
-     * @brief Registers the given \p callback that is called when a reservation should be cancelled.
-     *
-     * @param callback
-     */
-    void register_reservation_cancelled_callback(const std::function<void(const int& connector_id)>& callback);
+    ///
+    /// \brief Constructor.
+    ///
+    ReservationHandler(std::map<int, std::unique_ptr<module::EVSEContext>>& evses, std::recursive_mutex& evse_mutex,
+                       const std::string& id, kvsIntf* store);
+
+    ///
+    /// \brief Destructor.
+    ///
+    ~ReservationHandler();
+
+    ///
+    /// \brief Initialize. Read reservations from persistent store.
+    ///
+    void init();
+
+    ///
+    /// \brief Try to make a reservation.
+    /// \param evse_id      Optional, the evse id. If omitted, a 'global' reservation will be made.
+    /// \param reservation  The reservation to make.
+    /// \return The result of the reservation (`Accepted` if the reservation could be made).
+    ///
+    types::reservation::ReservationResult make_reservation(const std::optional<uint32_t> evse_id,
+                                                           const types::reservation::Reservation& reservation);
+
+    ///
+    /// \brief Change the EVSE state.
+    ///
+    /// This is important for the reservation handler, to know which EVSE is in which state, to know if a reservation
+    /// can be made or not.
+    ///
+    /// \param available    True if evse is now available.
+    /// \param evse_id      The EVSE id.
+    ///
+    void on_evse_available_changed(const bool available, const uint32_t evse_id);
+
+    ///
+    /// \brief Change a specific connector state.
+    ///
+    /// This is important for the reservation handler, to know which connector is in which state, to know if a
+    /// reservation can be made or not.
+    ///
+    /// \param connector_state  The state of the connector.
+    /// \param evse_id          The EVSE id the connector belongs to.
+    /// \param connector_id     The connector id.
+    ///
+    void on_connector_state_changed(const ConnectorState connector_state, const uint32_t evse_id,
```

Comment:
You don't need to pass the `connector_state` here since the `ReservationManager` has a reference the the `evse` map and can check the status internally